protos/controllers.proto

```diff
@@ -0,0 +1,15 @@
+syntax = "proto3";
+
+package dcs;
+
+message SetAlarmStateRequest {
+	enum AlarmState {
+		AUTO = 0;
+		GREEN = 1;
+		RED = 2;
+	}
+
+	optional string group_name = 1;
+	optional string unit_name = 2;
```

Comment:
Would a `oneof` be better here? (might also fix the compile error)

Reply:
God I am an idiot. Yes that is the _obvious_ thing to do and it bypasses the issue above completely.